plugins/inputs/tail/tail.go

```diff
@@ -229,21 +268,18 @@ func (t *Tail) receiver(parser parsers.Parser, tailer *tail.Tail) {
 
 		for _, metric := range metrics {
 			metric.AddTag("path", tailer.Filename)
-			t.acc.AddMetric(metric)
 		}
-	}
-
-	t.Log.Debugf("Tail removed for %q", tailer.Filename)
 
-	if err := tailer.Err(); err != nil {
-		t.Log.Errorf("Tailing %q: %s", tailer.Filename, err.Error())
+		select {
+		case <-t.ctx.Done():
```

Comment:
I think it's possible for this plugin to have the context signal that it's cancelled and then the plugin doesn't exit because it's tailing a file with very infrequent updates. Probably don't need to do anything about this, but I figured I'd point it out.

Reply:
In Stop() we close the Tailer, this should make tailer.Lines return immediately.